src/domains/product/utils/transformFilterValueToQueryString.ts

```
@@ -4,30 +4,58 @@ import {
+const excludeInitFilterValue = (query: Partial<IFilterType>): Partial<IFilterType> =>
+  Object.keys(query)
+    .filter((key) => {
+      const k = key as keyof IFilterType;
+      const value = query[k];
+      const initValue = INIT_FILTER_VALUE[k];
+      const isArray = Array.isArray(value) && Array.isArray(initValue);
+      return isArray ? !isEqualArray(value, initValue) : value !== initValue;
+    })
+    .reduce((acc, key) => {
+      const k = key as keyof IFilterType;
+      return { ...acc, [k]: query[k] };
```

Biome

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)